src/Kucipong/Db/Models/Base.hs

```diff
@@ -275,30 +468,22 @@ instance PersistFieldSql BusinessCategoryDetail where
 
 instance ToJSON BusinessCategoryDetail where
   toJSON :: BusinessCategoryDetail -> Value
-  toJSON (GourmetDetail a) = toJSON a
-  toJSON (FashionDetail a) = toJSON a
-  toJSON (GadgetDetail a) = toJSON a
-  toJSON (TravelingDetail a) = toJSON a
-  toJSON (BeautyDetail a) = toJSON a
-  toJSON (CommonDetail a) = toJSON a
+  toJSON = foldAllBusinessCategoryDetail (Proxy :: Proxy ToJSON) toJSON
 
 instance FromJSON BusinessCategoryDetail where
   parseJSON x =
-    case mdetail of
-      Nothing ->
-        typeMismatch
-          ("Tried to convert \"" <> show x <>
-           "\" to business category detail, but failed.")
-          x
-      Just detail -> pure detail
+    fromMaybeOrM mdetail $
+    typeMismatch
+      ("Tried to convert \"" <> show x <>
+       "\" to business category detail, but failed.")
+      x
     where
+      mdetail :: Maybe BusinessCategoryDetail
       mdetail =
-        (GourmetDetail <$> parseMaybe parseJSON x) <|>
-        (FashionDetail <$> parseMaybe parseJSON x) <|>
-        (GadgetDetail <$> parseMaybe parseJSON x) <|>
-        (TravelingDetail <$> parseMaybe parseJSON x) <|>
-        (BeautyDetail <$> parseMaybe parseJSON x) <|>
-        (CommonDetail <$> parseMaybe parseJSON x)
+        unfoldAllBusinessCategoryDetailAlt
```

Comment:
This is slightly easier to write, and we don't have to worry about missing a case.

For example, if a new constructor `SportsDetail` was added to `BusinessCategoryDetail`, we would have to remember to update `parseJSON`, `businessCategoryDetailFromText`, etc.  However, if we use `unfoldAllBusinessCategoryDetailAlt`, all we have to remember is to update the `unfoldBusinessCategory` function